### ISSUE TITLE
SQL Syntax highlighting support

### ISSUE THREAD

**Issue by ijzerbroot** [zed-industries/feedback#342]
**Is your feature request related to a problem? Please describe.**
Nope

**Describe the solution you'd like**
(basic) SQL syntax highlighting would be a quality of life improvement feature for me.
I imagine it would/should be way down on the list of things to add but hey, one can always ask :)

Attached is an example of how neovim does it (which is imperfect but gives the general idea)
In practice to me it would mainly be sensible to colorize:
- Reserved keywords (INSERT, UPDATE, DELETE, BEGIN, CREATE, ALTER, DROP, FROM, SELECT, UNION, etc.)
- Quoted strings
- Numeric values
- Comments
- Data types (text, integer, biting, float, serial, numeric, varchar, etc).

**Describe alternatives you've considered**
N/A

**Screenshots**
<img width="638" alt="Screenshot 2022-07-17 at 11 39 35" src="https://user-images.githubusercontent.com/27082653/179392635-1e18b44b-a499-45ae-a59c-748a92b59539.png">

**Comment by evoshy** [comment]
@hovsater i'd urge you to not propose this as a plugin, SQL syntax highlighting should be baked right into the editor and anything else related to SQL or databases in general could be a plugin imo, do us backend devs a favor of at least adding syntax highlighting because i'd love to have direct support for the syntax when i want to view some schema in SQL.

**Comment by kwando** [comment]
Yeah, just syntax highlighting would be an huge improvement :)

**Comment by krstp** [comment]
Longing for this feature big time.

**Comment by oliver-kriska** [comment]
Any ETA for SQL syntax highlighting please? :)

**Comment by paganotoni** [comment]
FWIW: This is one of my show-stoppers for using Zed. It would be huge to have SQL syntax highlighted.

**Comment by JosephTLyons** [comment]
> I wonder if the Zed team only uses tree sitter based highlighting? If so, there are examples for SQL such as https://github.com/DerekStride/tree-sitter-sql

This is correct. Zed relies solely on tree sitter for syntax highlighting. We will need to use a tree sitter grammar for SQL to add support for it.
